Content.Client/Examine/ExamineSystem.cs

```diff
@@ -133,69 +188,151 @@ public async void DoExamine(EntityUid entity)
-                try
-                {
-                    _requestCancelTokenSource = new CancellationTokenSource();
-                    response =
-                        await AwaitNetworkEvent<ExamineSystemMessages.ExamineInfoResponseMessage>(_requestCancelTokenSource
-                            .Token);
-                }
-                catch (TaskCanceledException)
-                {
-                    return;
-                }
-                finally
-                {
-                    _requestCancelTokenSource = null;
-                }
-
-                message = response.Message;
+            SortedSet<Verb>? set = null;
+            if (getVerbs)
+            {
+                // Get verbs
+                var verbs = _verbSystem.GetVerbs(target, player, VerbType.Examine);
+                verbs.TryGetValue(VerbType.Examine, out set);
             }
 
             foreach (var msg in message.Tags.OfType<FormattedMessage.TagText>())
             {
                 if (string.IsNullOrWhiteSpace(msg.Text)) continue;
 
-                var richLabel = new RichTextLabel();
+                var richLabel = new RichTextLabel() { Margin = new Thickness(4, 4, 0, 4)};
                 richLabel.SetMessage(message);
                 vBox.AddChild(richLabel);
                 break;
             }
+
+            if (set != null)
+            {
+                AddVerbsToTooltip(set);
+            }
+        }
+
+        private void AddVerbsToTooltip(IEnumerable<Verb> verbs)
+        {
+            if (_examineTooltipOpen == null)
+                return;
+
+            var buttonsHBox = new BoxContainer
+            {
+                Name = "ExamineButtonsHBox",
+                Orientation = LayoutOrientation.Horizontal,
+                HorizontalAlignment = Control.HAlignment.Right,
+                VerticalAlignment = Control.VAlignment.Bottom,
+            };
+
+            // Examine button time
+            foreach (var verb in verbs)
+            {
+                if (verb.IconTexture == null)
+                    continue;
+
+                // snowflake this for now.
+                if (verb.Text == "Basic")
+                    continue;
+
+                var button = new ExamineButton(verb);
+
+                button.OnPressed += VerbButtonPressed;
+                buttonsHBox.AddChild(button);
+            }
+
+            var vbox = _examineTooltipOpen?.GetChild(0).GetChild(0);
+            if (vbox == null)
+            {
+                buttonsHBox.Dispose();
+                return;
+            }
+
+            // Remove any existing buttons hbox, in case we generated it from the client
+            // then received ones from the server
+            var hbox = vbox.Children.Where(c => c.Name == "ExamineButtonsHBox").ToArray();
+            if (hbox.Any())
+            {
+                vbox.Children.Remove(hbox.First());
+            }
```

Comment:
this is also very sus but again can't think of a better way to do it

Reply:
I guess if you make the `_examineTooltipOpen` a xaml class, you can just set the "ExamineButtonsHBox"'s visibility and directly call `DisposaAllChildren()` on it or whatever its called

Reply:
i might be doing further stuff with examine so i'll XAML it in a future PR i think